src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.MaxNumber.cs

```diff
@@ -0,0 +1,142 @@
+            where T : INumber<T> =>
+            InvokeSpanSpanIntoSpan<T, MaxNumberOperator<T>>(x, y, destination);
+
+        /// <summary>Computes the element-wise maximum of the numbers in the specified tensors.</summary>
+        /// <param name="x">The first tensor, represented as a span.</param>
+        /// <param name="y">The second tensor, represented as a scalar.</param>
+        /// <param name="destination">The destination tensor, represented as a span.</param>
+        /// <exception cref="ArgumentException">Destination is too short.</exception>
+        /// <exception cref="ArgumentException"><paramref name="x"/> and <paramref name="destination"/> reference overlapping memory locations and do not begin at the same location.</exception>
+        /// <remarks>
+        /// <para>
+        /// This method effectively computes <c><paramref name="destination" />[i] = <typeparamref name="T"/>.MaxNumber(<paramref name="x" />[i], <paramref name="y" />)</c>.
+        /// </para>
+        /// <para>
+        /// The determination of the maximum element matches the IEEE 754:2019 `maximumNumber` function. If either value is <see cref="IFloatingPointIeee754{TSelf}.NaN"/>
+        /// the other is returned. Positive 0 is considered greater than negative 0.
+        /// </para>
+        /// <para>
+        /// This method may call into the underlying C runtime or employ instructions specific to the current architecture. Exact results may differ between different
+        /// operating systems or architectures.
+        /// </para>
+        /// </remarks>
+        public static void MaxNumber<T>(ReadOnlySpan<T> x, T y, Span<T> destination)
+            where T : INumber<T> =>
+            InvokeSpanScalarIntoSpan<T, MaxNumberOperator<T>>(x, y, destination);
+
+        /// <summary>T.MaxNumber(x, y)</summary>
+        internal readonly struct MaxNumberOperator<T> : IAggregationOperator<T> where T : INumber<T>
+        {
+            public static bool Vectorizable => true;
+
+            public static T Invoke(T x, T y) => T.MaxNumber(x, y);
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static Vector128<T> Invoke(Vector128<T> x, Vector128<T> y)
+            {
+                if (typeof(T) == typeof(float) || typeof(T) == typeof(double))
+                {
+                    if (AdvSimd.IsSupported && typeof(T) == typeof(float))
+                    {
+                        return AdvSimd.MaxNumber(x.AsSingle(), y.AsSingle()).As<float, T>();
+                    }
+
+                    if (AdvSimd.Arm64.IsSupported && typeof(T) == typeof(double))
+                    {
+                        return AdvSimd.Arm64.MaxNumber(x.AsDouble(), y.AsDouble()).As<double, T>();
+                    }
+
+                    return
+                        Vector128.ConditionalSelect(Vector128.Equals(x, y),
+                            Vector128.ConditionalSelect(IsNegative(y), x, y),
```

Comment:
I'm guessing this expression encodes the `NaN` filtering behaviour, but I'm not entirely sure what this second line is doing. My intuition tells me that this could be either `x` or `y`, but perhaps I'm missing some corner case?

Reply:
Perhaps it's for filtering out `-0.0`? But couldn't `x` also be `-0.0` in that case?

Reply:
The outermost `Vector128.ConditionalSelect(Vector128.Equals(x, y), a. b)` is saying pick `a` if `x` and `y`are equal and `b` otherwise.

For `a`, we're then doing `Vector128.ConditionalSelect(IsNegative(y), x, y)`.  Since `x` and `y` are equal, they are either `0` of different signs or bitwise identical values (in which case which is returned doesn't matter). If `x: +0.0; y: -0.0`, then `IsNegative(y)` is true and we return `x`. Otherwise, we must have `x: -0.0; y: +0.0`, `IsNegative(y)` is false and we return `y`. Both scenarios pick the `+0.0` when the signs differ, and otherwise they're bitwise identical so it doesn't matter.

For `b`, we're then doing `Vector128.ConditionalSelect(Vector128.Equals(y, y), Vector128.Max(x, y), x))`. Since `x` and `y` are not equal, they are either completely different values or both `NaN`. If: `x: non-NaN; y: non-NaN`, then `Equals(y, y)` is `true` and `Max(x, y)` picks the right result. If `x: non-NaN; y: NaN`, then `Equals(y, y)` is false and we return `x` (the number). If `x: NaN; y: non-NaN`, then `Equals(y, y)` is true and `Max(x, y)` will return `y` (this is the documented behavior of the `maxps` instruction for x86/x64). Otherwise, `x: NaN; y: NaN`, in which case `Equals(y, y)` is false and we return `x` (which of the two NaN we return doesn't really matter, we just happen to pick `x`).

-- The only notable thing is that the handling of `NaN` for `Vector128.Max(x, y)` isn't strictly defined across platforms today. We're working on fixing that so that it behaves identically to `float.Max`, but today its platform specific. So while this behaves as expected for all platforms that can hit the code path today, it's potentially something we need to adjust in the future (it would fail if Arm32 acceleration were brought online, for example).

Reply:
> For a, we're then doing Vector128.ConditionalSelect(IsNegative(y), x, y). Since x and y are equal, they are either 0 of different signs or bitwise identical values (in which case which is returned doesn't matter). If x: +0.0; y: -0.0, then IsNegative(y) is true and we return x. Otherwise, we must have x: -0.0; y: +0.0,

Why does it follow that they have different signs? -0.0 equals -0.0 and in that case the output would still be -0.0.

Reply:
> Why does it follow that they have different signs?

Because `-0.0 == +0.0` (thus `Equals` reports them as the same), but IEEE 754 spec requires that `Min`, `MinNumber`, `Max`, and `MaxNumber` treat it as though `-0.0 < +0.0` (hence why this secondary condition exists)

Reply:
Would it make sense to add a comment explaining that this is here to guarantee `-0.0 < +0.0` ordering? Should there be tests for this?

Reply:
> Should there be tests for this?

We have some, e.g. via https://github.com/dotnet/runtime/blob/74d927967dcf9dc1801c310603d6dc843dc28ad6/src/libraries/System.Numerics.Tensors/tests/TensorPrimitives.Generic.cs#L335-L338, but we could separately add some more if we still have gaps.

Reply:
We also have fairly comprehensive tests covering this for scalar values. One such example is  https://github.com/dotnet/runtime/blob/main/src/libraries/System.Runtime/tests/System.Runtime.Tests/System/SingleTests.cs#L1061, but others exist spread out across Math, MathF, and SingleTests.GenericMath, and similar for types like double, half, nfloat, etc